src/rt-app.c

```diff
@@ -686,6 +670,40 @@ void *thread_body(void *arg)
 			exit(EXIT_FAILURE);
 	}
 
+	data->current_prio = sched_prio;
+}
+
+void *thread_body(void *arg)
+{
+	thread_data_t *data = (thread_data_t*) arg;
+	phase_data_t *pdata;
+	log_data_t ldata;
+	struct sched_param param;
```

Comment:
I understand you would like to get deadline support in a second step, but don't we break deadline here?
IIUC, this is the sched_param param thread_body uses below to set deadline attributes, but set_thread_priority() operates on a different (local) sched_param param, so the one we actually use for the syscall might be uninitialized?